src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+import java.io.File;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.ArrayList;
+
 public class Duke {
+    private SaveToFile saveToFile;
+    private Ui ui;
+    private TaskList tasks;
+
+    /**
+     * Constructor for duke.
+     *
+     * @param filePath File path for duke's task list.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        saveToFile = new SaveToFile();
+        ArrayList<Task> arr = new ArrayList<>();
+        tasks = new TaskList(saveToFile.loadList(filePath));
+    }
+
+    /**
+     * Run duke
+     */
+    public void run() {
+        ui.showWelcome();
+        Scanner sc = new Scanner(System.in);
+
+        while (sc.hasNext()) {
+            String userInput = sc.nextLine();
+            String[] arrString = userInput.split(" ", 2);
+            try {
+                if (arrString[0].equalsIgnoreCase("bye")) {
+                    ui.goodbyeMessage();
+                    break;
+                } else if (arrString[0].equalsIgnoreCase("list")) {
+                    ui.showLine();
+                    if (tasks.getTaskListSize() == 0) {
+                        ui.emptyList();
+                    } else {
+                        StringBuilder sb = new StringBuilder();
+                        for (int i = 0; i < tasks.getTaskListSize(); i++) {
+                            System.out.println(i + 1 + ". " + tasks.getTask(i).toString());
+                            sb.append(tasks.getTask(i).saveToList() + "\n");
+                        }
+                        saveToFile.usingFileWriter(sb.toString());
+                        ui.showLine();
+                    }
+                } else if (arrString[0].equalsIgnoreCase("done")) {
```

Comment:
I like how you used .equalsIgnoreCase to ensure that varied user input is captured as well.